mm2src/coins/my_tx_history_v2.rs

```diff
@@ -210,15 +213,8 @@ impl<'a, Addr: Clone + DisplayAddress + Eq + std::hash::Hash, Tx: Transaction> T
                 bytes_for_hash.extend_from_slice(&token_id.0);
                 sha256(&bytes_for_hash).to_vec().into()
             },
-            TransactionType::TendermintIBCTransfer { token_id }
-            | TransactionType::CustomTendermintMsg { token_id, .. } => {
-                if let Some(token_id) = token_id {
-                    let mut bytes_for_hash = tx_hash.0.clone();
-                    bytes_for_hash.extend_from_slice(&token_id.0);
-                    sha256(&bytes_for_hash).to_vec().into()
-                } else {
-                    tx_hash.clone()
-                }
+            TransactionType::TendermintIBCTransfer { .. } | TransactionType::CustomTendermintMsg { .. } => {
+                unreachable!("Tendermint never invokes this function.")
```

Comment:
i see this method is infallible. could we write a correct implementation for this if that's feasible, instead of panicking.

Reply:
There is the TODO note I wrote on top of this function.

Reply:
it's not targeting this panic speifically tho. and who knows when this todo would be picked up.
if we can easily impl this (but only if), for my money, we better do as we lose nothing.

Reply:
It's covering it. If you remove this function and implement it in the protocol details you won't have to do pattern matching on all the protocols.